test/unit/realm_test.rb

```diff
@@ -0,0 +1,87 @@
+require File.join(File.dirname(__FILE__), 'test_helper')
+require File.join(File.dirname(__FILE__), 'apipie_resource_mock')
+
+require 'hammer_cli_foreman/realm'
+
+describe HammerCLIForeman::Realm do
+
+  include CommandTestHelper
+
+  context "ListCommand" do
+
+    let(:cmd) { HammerCLIForeman::Realm::ListCommand.new("", ctx) }
+
+    context "parameters" do
+      it_should_accept "no arguments"
+      it_should_accept_search_params
+    end
+
+    context "output" do
+      let(:expected_record_count) { count_records(cmd.resource.call(:index)) }
+
+      it_should_print_n_records
+      it_should_print_columns ["Id", "Name"]
+    end
+
+  end
+
+
+  context "InfoCommand" do
+
+    let(:cmd) { HammerCLIForeman::Realm::InfoCommand.new("", ctx) }
+
+    context "parameters" do
+      it_should_accept "id", ["--id=1"]
+      it_should_accept "name", ["--name=arch"]
+      # it_should_fail_with "no arguments" # TODO: temporarily disabled, parameters are checked in the id resolver
+    end
+
+    context "output" do
+      with_params ["--id=1"] do
+        it_should_print_n_records 1
+        it_should_print_columns ["Id", "Name", "Created at", "Updated at"]
+        it_should_print_columns ["Realm proxy id", "Realm type"]
+      end
+    end
+
+  end
+
+  context "CreateCommand" do
+
+    let(:cmd) { HammerCLIForeman::Realm::CreateCommand.new("", ctx) }
+
+    context "parameters" do
+      it_should_accept "name, type, proxy", ["--name=realm", "--realm-type=FreeIPA", "--realm-proxy-id=1"]
```

Comment:
These kinsd of tests are commented out in the others because they don't work, for example, see the user_test: https://github.com/theforeman/hammer-cli-foreman/blob/master/test/unit/user_test.rb#L58-L62